### ISSUE TITLE
Board editor: Assert is raised when starting trace at unconnected pad

### ISSUE THREAD

**Issue by ubruhin** [LibrePCB/LibrePCB#762]
##### VERSION / OS / ENVIRONMENT

```
LibrePCB Version: 0.2.0-unstable
Git Revision:     2040eac1afbc0cf1f135884d9c05bf62b156a21c
Build Date:       2020-08-31T20:59:09
Linking type:     static
Unbundled libs:   -
Qt Version:       5.5.1 (built against 5.5.1)
CPU Architecture: x86_64
Operating System: Ubuntu 16.04.7 LTS
Platform Plugin:  xcb
```

##### SUMMARY

When starting to draw a trace at a footprint pad which has no netsignal, an assert is raised (resp. crashes in release mode).

##### STEPS TO REPRODUCE

- Add a device to a board containing pads which are not connected
- Start the "draw trace" tool
- Click on the unconnected pad

##### EXPECTED RESULTS

The trace should start at the pad.

##### ACTUAL RESULTS

The following assert is raised:

https://github.com/LibrePCB/LibrePCB/blob/1f2c0d411a165ae42772b49549dbef78a3c91794/libs/librepcb/projecteditor/boardeditor/fsm/bes_drawtrace.cpp#L566

/cc @5n8ke

**Comment by 5n8ke** [comment]
I'll take a look

**Comment by 5n8ke** [comment]
Done. There was an exception missing to check for unconnected pads. Every other item used for drawing traces shouldn't/can't exist without a netsignal.

**Comment by 5n8ke** [comment]
@ubruhin On closer inspection this morning, I'm actually not sure how this bug can be reproduced with the current master version since findPad() does only return pads that are connected to a NetSignal.
This does result in the already present error "Nothing to connect here".

Could you supply me with a save that has this issue?

**Comment by ubruhin** [comment]
Whooops, it seems I mixed up things :scream: Probably I checked out #753 locally when I experienced this, and thought it also happens on `master` since it also crashed when clicking *on* a pad, not just when clicking *beside* it :sob: I didn't realize that clicking *on* the pad still triggers `findAnchorNextTo()`. So the fix in #753 seems to be all we need.

Sorry for the confusion! :slightly_smiling_face:
